pkg/graph/resolvers/mto_solution.go

```diff
@@ -0,0 +1,152 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+
+	"github.com/cms-enterprise/mint-app/pkg/authentication"
+	"github.com/cms-enterprise/mint-app/pkg/models"
+	"github.com/cms-enterprise/mint-app/pkg/sqlutils"
+	"github.com/cms-enterprise/mint-app/pkg/storage"
+	"github.com/cms-enterprise/mint-app/pkg/storage/loaders"
+)
+
+// MTOSolutionGetByModelPlanIDLOADER implements resolver logic to get all MTO
+// solutions by a model plan ID using a data loader
+func MTOSolutionGetByModelPlanIDLOADER(
+	ctx context.Context,
+	modelPlanID *uuid.UUID,
+) ([]*models.MTOCommonSolution, error) {
+
+	// Translate a nil key to UUID nil, as we need a primitive type for translating results later
+	var key uuid.UUID
+	if modelPlanID != nil {
+		key = *modelPlanID
+	}
+	return loaders.MTOCommonSolution.ByModelPlanID.Load(ctx, key)
+}
+
+func MTOSolutionGetCommonSolutionByKeyLoader(
+	namedPreparer sqlutils.NamedPreparer,
+	logger *zap.Logger,
+	key *models.MTOCommonSolutionKey,
+) (*models.MTOCommonSolution, error) {
+	if key == nil {
+		return nil, fmt.Errorf("common solution key is nil")
+	}
+
+	commonSolution, err := storage.MTOCommonSolutionGetByKeyLoader(
+		namedPreparer,
+		logger,
+		[]models.MTOCommonSolutionKey{*key},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(commonSolution) == 0 {
+		return nil, fmt.Errorf("common solution not found for solution key %s", *key)
+	}
+
+	return commonSolution[0], nil
+}
+
+// MTOSolutionUpdate updates the MTOSolution
+func MTOSolutionUpdate(
+	logger *zap.Logger,
+	principal authentication.Principal,
+	store *storage.Store,
+	id uuid.UUID,
+	changes map[string]interface{},
+) (*models.MTOSolution, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+	existing, err := storage.MTOSolutionGetByIDLoader(store, logger, []uuid.UUID{id})
```

Comment:
In order to keep things moving forward I'm going to open the PR while I figure out how to address this piece of feedback.

Reply:
I'm alright to flag this with a `TODO` to revisit naming for now, but to just share my immediate thoughts:

This storage function isn't doing anything that's technically loader specific. It just says "Give me a list of IDs, and I'll return MTO Solutions for each of those IDs"

If that's used by a loader, that makes sense, but there's not any reason I'd say a function like this needs to be specifically designed or named around the loaders if it can be used in other contexts (like this one, technically!)

Let's sync on this sometime next week and see if we can approach naming / designing these in a way that feels more clear, but I'd rather just flag this with a comment for now and move past it for the sake of this PR 😁

Reply:
@OddTomBrooks, distinct from the naming convention of the store method, can you please update this to call a data loader method? 

We landed on using the data loader in the update resolver is acceptable (and preferred so the data is batched if called elsewhere).